tensorboard/components/tf_dashboard_common/data-loader-behavior.ts

```diff
@@ -146,40 +142,39 @@ namespace tf_dashboard_common {
       }
     },
 
-    _loadDataImpl() {
+    async _loadDataImpl() {
       if (!this.active) return;
-      this.cancelAsync(this._loadDataAsync);
-      this._loadDataAsync = this.async(() => {
-        // Read-only property have a special setter.
-        this._setDataLoading(true);
 
-        // Before updating, cancel any network-pending updates, to
-        // prevent race conditions where older data stomps newer data.
-        this._canceller.cancelAll();
+      // Before updating, cancel any network-pending updates, to
+      // prevent race conditions where older data stomps newer data.
+      this._canceller.cancelAll();
+
+      // Read-only property have a special setter.
+      this._setDataLoading(true);
+
+      const update = this._canceller.cancellable(async (result) => {
         const promises = this.dataToLoad
           .filter((datum) => {
             const name = this.getDataLoadName(datum);
             return !this._loadedData.has(name);
           })
-          .map((datum) => {
+          .map(async (datum) => {
```

Comment:
I personally prefer not using async here because it is harder to understand which promise this is referring to.